src/layers/XRLayer/XRLayer.js

```diff
@@ -55,11 +55,13 @@ function getRenderingAttrs(dtype, gl, interpolation) {
   upgradedShaderModule.fs = version300str.concat(upgradedShaderModule.fs);
   upgradedShaderModule.vs = version300str.concat(upgradedShaderModule.vs);
   const values = getDtypeValues(isLinear ? 'Float32' : dtype);
+  // Check if a cast function is present in the returned data for the dtype
+  const cast = values.cast || (data => data);
```

Comment:
Can you extract the inline function to a separate util? And then maybe a more strict guard?

```javascript
const identity = x => x;

//

const case = typeof values.cast === "function" ? values.cast : identity;
```

Reply:
Done!